### PR TITLE
Send notification to user when he's involved in an action

### PR DESCRIPTION
**Pull request by TheoGauchoux**
See the API PR : https://github.com/awi2017-option1group3/Prello-API/pull/38

Notification system: when an action is done by someone and if the action involved another user (assigned to a card or whatever), then send him a notification (including the user who did the action, the moment, a message and the board concerned).

List of actions sending a notification:

* The user has been added as contributor of a board
* The user has been removed from a board
* A board including the user has been renamed
* A board including the user has been deleted
* The user has been assigned as responsible / assignee to a card
* The user has been removed from a card
* A card including the user has been renamed
* A card including the user has been deleted
* A card including the user has been moved to another list
* A card including the user has been renamed
* The responsible of a card where the user is just assignee has been updated

In the future, we need to include (when the features will be available):
* The due date of a card including the user has been updated
* A comment has been post in a card including the user

### CODE REVIEW

File: src/components/assignees/ResponsibleSelect.jsx [hunk 1]
```diff
@@ -14,40 +14,62 @@ class ResponsibleSelect extends Component {
...
     }
     this.setState({
-      placeholder: 'Choose responsible...'
+      placeholder: 'Choose responsible...',
     })
   }
 
   render() {
-    return (
+    return this.props.responsible ? (
```
Comment (Tombean) [inline src/components/assignees/ResponsibleSelect.jsx]: remove this condition. You do want to be able to select a responsible if there is no responsible yet

File: src/components/assignees/ResponsibleSelect.jsx [hunk 2]
```diff
@@ -14,40 +14,62 @@ class ResponsibleSelect extends Component {
...
         className="cardResponsible"
         placeholder={this.state.placeholder}
         onChange={this.handleChangeResponsible}
         tokenSeparators={[',']}
       >
-        { this.props.allUsers.map(user => <Option key={this.props.cardId + user.id}>{ user.fullName }</Option>) }
+        { this.props.allUsers.map(user => <Option key={this.props.card.id + user.id}>{ user.fullName }</Option>) }
       </Select>
-    )
+    ) : (null)
```
Comment (Tombean) [inline src/components/assignees/ResponsibleSelect.jsx]: See above :  remove condition

File: src/components/notifications/Notifications.jsx [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+      </div>
+    )
+  }
+
+  renderContent() {
+    return this.props.notifications.length > 0 ? (
+      <div className="notificationsList">
+        {this.props.notifications
+          .filter(notification => !notification.isRead)
+          .sort((a, b) => moment(b.occuredAt) - moment(a.occuredAt))
```
Comment (Tombean) [inline src/components/notifications/Notifications.jsx:56]: why sort on a difference and not on : 
.sort((a, b) => moment(b.occuredAt) > moment(a.occuredAt))

File: src/containers/board/BoardContainer.jsx [hunk 4]
```diff
@@ -19,10 +21,12 @@ class BoardContainer extends Component {
   }
 
   render() {
-    if (this.props.board.isFailed) {
-      return (<ErrorDisplayer message="Failed to load this board. Maybe it doesn't exist, or you don't have access to it." />)
-    } else if (this.props.user) {
-      return (<DndBoard {...this.props} {...this.props.board} boardId={this.props.match.params.boardId} />)
+    if (this.props.user) {
+      return this.props.board.isFailed ? (
```
Comment (Tombean) [inline src/containers/board/BoardContainer.jsx]: Please change : isFailed -> hasFailed

**Review by Tombean — CHANGES_REQUESTED** [review]

File: src/components/notifications/Notifications.jsx (hunk 3 above)
Comment (TheoGauchoux) [reply src/components/notifications/Notifications.jsx:56]: Sort accept only integer to arrange according to the sign (negative / positive)

**Review by Tombean — APPROVED** [review]

**Pull request merged**